lm_eval/api/task.py

```diff
@@ -435,12 +438,22 @@ def build_all_requests(
             total=num_docs,
         ):
             # sample fewshot context #TODO: need to offset doc_id by rank now!
+            doc_system_instruction = system_instruction or ""
+            if self.OUTPUT_TYPE == "multiple_choice" and multiple_choice_generate:
+                if doc_system_instruction:
+                    doc_system_instruction += " "
+                if multiple_choice_generate == "abcd":
```

Comment:
May I suggest to not hardcode these. What if doc_system_instruction supposed to be delimited with some other delimiter? What if set of choices is not 4 letters, not these 4 letters, or not letters at all? This framework supports external tasks and also have multiple forks already, so there may be (I am not using "are" because of no intention to google proof of this idea) multiple choice tasks set up differently than "abcd".